t/33-vagrant.t

```diff
@@ -0,0 +1,511 @@
+#!/usr/bin/perl
+
+use Cwd qw(abs_path);
+use Test::Most;
+use File::chdir;
+use FindBin '$Bin';
+use lib "$Bin/../external/os-autoinst-common/lib";
+use OpenQA::Test::TimeLimit '20';
+use Test::MockRandom 'backend::vagrant';
+use Test::MockObject;
+use Test::MockModule;
+use Test::Warnings ':report_warnings';
+use Test::Output qw(stderr_like);
+use backend::vagrant;
+
+use bmwqemu ();
+
+# hide debug messages, keep warnings & errors
+$bmwqemu::logger = Mojo::Log->new(level => 'warn');
+
+my $mock_vagrant = Test::MockModule->new("vagrant", no_auto => 1);
+$mock_vagrant->noop('mkdir');
+
+my $tmpdir        = File::Temp->newdir();
+my $mock_file_tmp = Test::MockModule->new("File::Temp");
+$mock_file_tmp->redefine('newdir', sub { return $tmpdir; });
+
+my $expected_console_name        = undef;
+my $expected_console_type        = undef;
+my %expected_console_credentials = ();
+my $distri                       = Test::MockModule->new('distribution');
+$distri->redefine('add_console', sub {
+        my ($self, $name, $type, $creds) = @_;
+
+        if (defined($expected_console_name)) {
+            is($name, $expected_console_name, 'Check the name of the console');
```

Comment:
Could you point me to an existing example what you envision such a function to look like?

Reply:
Well, as I said, how about a `die`-call? So

```suggestion
            die "Invalid name of console '$name'. Expected '$expected_console_name'" unless $name eq $expected_console_name;
```

Reply:
The test framework allows to set the level, so that the correct line will be reported:
https://metacpan.org/pod/Test::Builder#Test-style
```perl
sub helper {
    local $Test::Builder::Level = $Test::Builder::Level + 1;
    ok($x);
}
```